common/retry/retry_test.go

```diff
@@ -9,66 +9,64 @@ import (
 
 func TestRetry(t *testing.T) {
 	var calls int
-	fn := func() (shouldRetry bool, err error) {
+	fn := func() (shouldRetry bool) {
 		calls++
-		return calls < 5, nil
-	}
-
-	if err := Retry(fn); err != nil {
-		t.Fatal(err)
+		return calls < 5
 	}
 
+	Retry(fn)
 	assert.Equal(t, 5, calls)
 }
 
 func TestRetryError(t *testing.T) {
-	fn := func() (shouldRetry bool, err error) {
-		return false, fmt.Errorf("some error")
+	var err error
+	fn := func() (shouldRetry bool) {
+		err = fmt.Errorf("some error")
+		return false
 	}
 
-	if err := Retry(fn); err == nil {
+	Retry(fn)
+
+	if err == nil {
 		t.Fatal("Error was nil!")
 	}
+
 }
 
 func TestRetryOptions(t *testing.T) {
 	newOption := func() (Option, *int) {
 		var calls int
-		option := func() error {
+		option := func() bool {
 			calls++
-			return nil
+			return true
 		}
 
 		return option, &calls
 	}
 
 	var calls int
-	fn := func() (shouldRetry bool, err error) {
+	fn := func() (shouldRetry bool) {
 		calls++
-		return calls < 5, nil
+		return calls < 5
 	}
 
 	optionA, callsA := newOption()
 	optionB, callsB := newOption()
 
-	if err := Retry(fn, optionA, optionB); err != nil {
-		t.Fatal(err)
-	}
+	Retry(fn, optionA, optionB)
 
 	assert.Equal(t, 5, *callsA)
 	assert.Equal(t, 5, *callsB)
 }
 
 func TestRetryOptionError(t *testing.T) {
```

Comment:
These updates have lost the behavior the original test was trying to assert: that the value returned by an option was being honored by the `Retry` function. This needs to check that the option gets called, and that the retry stop executing once it returns false. The test name needs to change, too.